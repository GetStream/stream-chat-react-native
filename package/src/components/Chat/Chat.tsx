import React, { PropsWithChildren, useEffect, useState } from 'react';
import { Image, Platform } from 'react-native';

import type { Channel, StreamChat } from 'stream-chat';

import { useAppSettings } from './hooks/useAppSettings';
import { useCreateChatContext } from './hooks/useCreateChatContext';
import { useIsOnline } from './hooks/useIsOnline';
import { useMutedUsers } from './hooks/useMutedUsers';

import { useSyncDatabase } from './hooks/useSyncDatabase';

import { ChannelsStateProvider } from '../../contexts/channelsStateContext/ChannelsStateContext';
import { ChatContextValue, ChatProvider } from '../../contexts/chatContext/ChatContext';
import { useDebugContext } from '../../contexts/debugContext/DebugContext';
import { useOverlayContext } from '../../contexts/overlayContext/OverlayContext';
import { DeepPartial, ThemeProvider } from '../../contexts/themeContext/ThemeContext';
import type { Theme } from '../../contexts/themeContext/utils/theme';
import {
  DEFAULT_USER_LANGUAGE,
  TranslationProvider,
} from '../../contexts/translationContext/TranslationContext';
import { useStreami18n } from '../../hooks/useStreami18n';
import init from '../../init';

import { SDK } from '../../native';
import { QuickSqliteClient } from '../../store/QuickSqliteClient';
import type { DefaultStreamChatGenerics } from '../../types/types';
import { DBSyncManager } from '../../utils/DBSyncManager';
import type { Streami18n } from '../../utils/Streami18n';
import { version } from '../../version.json';

init();

export type ChatProps<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = Pick<ChatContextValue<StreamChatGenerics>, 'client'> &
  Partial<Pick<ChatContextValue<StreamChatGenerics>, 'ImageComponent'>> & {
    /**
     * When false, ws connection won't be disconnection upon backgrounding the app.
     * To receive push notifications, its necessary that user doesn't have active
     * websocket connection. So by default, we disconnect websocket connection when
     * app goes to background, and reconnect when app comes to foreground.
     */
    closeConnectionOnBackground?: boolean;
    /**
     * Enables offline storage and loading for chat data.
     */
    enableOfflineSupport?: boolean;
    /**
     * Instance of Streami18n class should be provided to Chat component to enable internationalization.
     *
     * Stream provides following list of in-built translations:
     * 1. English (en)
     * 2. Dutch (nl)
     * 3. ...
     * 4. ...
     *
     * Simplest way to start using chat components in one of the in-built languages would be following:
     *
     * ```
     * const i18n = new Streami18n('nl');
     * <Chat client={chatClient} i18nInstance={i18n}>
     *  ...
     * </Chat>
     * ```
     *
     * If you would like to override certain keys in in-built translation.
     * UI will be automatically updated in this case.
     *
     * ```
     * const i18n = new Streami18n('nl');
     *
     * i18n.registerTranslation('nl', {
     *  'Nothing yet...': 'Nog Niet ...',
     *  '{{ firstUser }} and {{ secondUser }} are typing...': '{{ firstUser }} en {{ secondUser }} zijn aan het typen...',
     * });
     *
     * <Chat client={chatClient} i18nInstance={i18n}>
     *  ...
     * </Chat>
     * ```
     *
     * You can use the same function to add whole new language.
     *
     * ```
     * const i18n = new Streami18n('it');
     *
     * i18n.registerTranslation('it', {
     *  'Nothing yet...': 'Non ancora ...',
     *  '{{ firstUser }} and {{ secondUser }} are typing...': '{{ firstUser }} a {{ secondUser }} stanno scrivendo...',
     * });
     *
     * // Make sure to call setLanguage to reflect new language in UI.
     * i18n.setLanguage('it');
     * <Chat client={chatClient} i18nInstance={i18n}>
     *  ...
     * </Chat>
     * ```
     */
    i18nInstance?: Streami18n;
    /**
     * You can pass the theme object to customize the styles of Chat components. You can check the default theme in [theme.ts](https://github.com/GetStream/stream-chat-react-native/blob/main/package/src/contexts/themeContext/utils/theme.ts)
     *
     * Please check section about [themes in cookbook](https://github.com/GetStream/stream-chat-react-native/wiki/Cookbook-v3.0#theme) for details.
     *
     * ```
     * import type { DeepPartial, Theme } from 'stream-chat-react-native';
     *
     * const theme: DeepPartial<Theme> = {
     *   messageSimple: {
     *     file: {
     *       container: {
     *         backgroundColor: 'red',
     *       },
     *       icon: {
     *         height: 16,
     *         width: 16,
     *       },
     *     },
     *   },
     * };
     *
     * <Chat style={theme}>
     * </Chat>
     * ```
     *
     * @overrideType object
     */
    style?: DeepPartial<Theme>;
  };

const ChatWithContext = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  props: PropsWithChildren<ChatProps<StreamChatGenerics>>,
) => {
  const {
    children,
    client,
    closeConnectionOnBackground = true,
    enableOfflineSupport = false,
    i18nInstance,
    ImageComponent = Image,
    style,
  } = props;

  const [channel, setChannel] = useState<Channel<StreamChatGenerics>>();

  // Setup translators
  const translators = useStreami18n(i18nInstance);

  /**
   * Setup connection event listeners
   */
  const { connectionRecovering, isOnline } = useIsOnline<StreamChatGenerics>(
    client,
    closeConnectionOnBackground,
  );

  const [initialisedDatabase, setInitialisedDatabase] = useState(false);

  /**
   * Setup muted user listener
   * TODO: reimplement
   */
  const mutedUsers = useMutedUsers<StreamChatGenerics>(client);

  const debugRef = useDebugContext();
  const isDebugModeEnabled = __DEV__ && debugRef && debugRef.current;

  useEffect(() => {
    if (client) {
      client.setUserAgent(`${SDK}-${Platform.OS}-${version}`);
      // This is to disable recovery related logic in js client, since we handle it in this SDK
      client.recoverStateOnReconnect = false;
      client.persistUserOnConnectionFailure = enableOfflineSupport;
    }

    if (isDebugModeEnabled) {
      if (debugRef.current.setEventType) debugRef.current.setEventType('send');
      if (debugRef.current.setSendEventParams)
        debugRef.current.setSendEventParams({
          action: 'Client',
          data: client.user,
        });
    }
  }, [client, enableOfflineSupport]);

  const setActiveChannel = (newChannel?: Channel<StreamChatGenerics>) => setChannel(newChannel);

  useEffect(() => {
    if (client.user?.id && enableOfflineSupport) {
      setInitialisedDatabase(false);
      QuickSqliteClient.initializeDatabase();
      DBSyncManager.init(client as unknown as StreamChat);
      setInitialisedDatabase(true);
    }
  }, [client?.user?.id, enableOfflineSupport]);

  const appSettings = useAppSettings(client, isOnline, enableOfflineSupport, initialisedDatabase);

  const chatContext = useCreateChatContext({
    appSettings,
    channel,
    client,
    connectionRecovering,
    enableOfflineSupport,
    ImageComponent,
    isOnline,
    mutedUsers,
    setActiveChannel,
  });

  useSyncDatabase({
    client,
    enableOfflineSupport,
  });

  return (
    <ChatProvider<StreamChatGenerics> value={chatContext}>
      <TranslationProvider
        value={{ ...translators, userLanguage: client.user?.language || DEFAULT_USER_LANGUAGE }}
      >
        <ThemeProvider style={style}>
          <ChannelsStateProvider<StreamChatGenerics>>{children}</ChannelsStateProvider>
        </ThemeProvider>
      </TranslationProvider>
    </ChatProvider>
  );
};

/**
 * Chat - Wrapper component for Chat. The needs to be placed around any other chat components.
 * This Chat component provides the ChatContext to all other components.
 *
 * The ChatContext provides the following props:
 *
 * - channel - currently active channel
 * - client - client connection
 * - connectionRecovering - whether or not websocket is reconnecting
 * - isOnline - whether or not set user is active
 * - setActiveChannel - function to set the currently active channel
 *
 * The Chat Component takes the following generics in order:
 * - At (AttachmentType) - custom Attachment object extension
 * - Ct (ChannelType) - custom Channel object extension
 * - Co (CommandType) - custom Command string union extension
 * - Ev (EventType) - custom Event object extension
 * - Me (MessageType) - custom Message object extension
 * - Re (ReactionType) - custom Reaction object extension
 * - Us (UserType) - custom User object extension
 */
export const Chat = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  props: PropsWithChildren<ChatProps<StreamChatGenerics>>,
) => {
  const { style } = useOverlayContext();

  return <ChatWithContext {...{ style }} {...props} />;
};
