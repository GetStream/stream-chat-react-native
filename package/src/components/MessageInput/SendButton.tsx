import React from 'react';

import { Pressable } from 'react-native';

import {
  MessageInputContextValue,
  useMessageInputContext,
} from '../../contexts/messageInputContext/MessageInputContext';
import { useTheme } from '../../contexts/themeContext/ThemeContext';
import { Search } from '../../icons/Search';
import { SendRight } from '../../icons/SendRight';
import { SendUp } from '../../icons/SendUp';

import type { DefaultStreamChatGenerics } from '../../types/types';

type SendButtonPropsWithContext<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = Pick<MessageInputContextValue<StreamChatGenerics>, 'giphyActive' | 'sendMessage'> & {
  /** Disables the button */ disabled: boolean;
};

const SendButtonWithContext = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  props: SendButtonPropsWithContext<StreamChatGenerics>,
) => {
  const { disabled = false, giphyActive, sendMessage } = props;
  const {
    theme: {
      colors: { accent_blue, grey_gainsboro },
      messageInput: { sendButton },
    },
  } = useTheme();

  return (
    <Pressable
      disabled={disabled}
      onPress={disabled ? () => null : sendMessage}
      style={[sendButton]}
      testID='send-button'
    >
      {giphyActive && <Search pathFill={disabled ? grey_gainsboro : accent_blue} />}
      {!giphyActive && disabled && <SendRight pathFill={grey_gainsboro} />}
      {!giphyActive && !disabled && <SendUp pathFill={accent_blue} />}
    </Pressable>
  );
};

const areEqual = <StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics>(
  prevProps: SendButtonPropsWithContext<StreamChatGenerics>,
  nextProps: SendButtonPropsWithContext<StreamChatGenerics>,
) => {
  const {
    disabled: prevDisabled,
    giphyActive: prevGiphyActive,
    sendMessage: prevSendMessage,
  } = prevProps;
  const {
    disabled: nextDisabled,
    giphyActive: nextGiphyActive,
    sendMessage: nextSendMessage,
  } = nextProps;

  const disabledEqual = prevDisabled === nextDisabled;
  if (!disabledEqual) return false;

  const giphyActiveEqual = prevGiphyActive === nextGiphyActive;
  if (!giphyActiveEqual) return false;

  const sendMessageEqual = prevSendMessage === nextSendMessage;
  if (!sendMessageEqual) return false;

  return true;
};

const MemoizedSendButton = React.memo(
  SendButtonWithContext,
  areEqual,
) as typeof SendButtonWithContext;

export type SendButtonProps<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = Partial<SendButtonPropsWithContext<StreamChatGenerics>>;

/**
 * UI Component for send button in MessageInput component.
 */
export const SendButton = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  props: SendButtonProps<StreamChatGenerics>,
) => {
  const { giphyActive, sendMessage } = useMessageInputContext<StreamChatGenerics>();

  return (
    <MemoizedSendButton
      {...{ giphyActive, sendMessage }}
      {...props}
      {...{ disabled: props.disabled || false }}
    />
  );
};

SendButton.displayName = 'SendButton{messageInput}';
