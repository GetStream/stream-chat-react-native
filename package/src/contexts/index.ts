export * from './attachmentPickerContext/AttachmentPickerContext';
export * from './channelContext/ChannelContext';
export * from './channelsContext/ChannelsContext';
export * from './chatContext/ChatContext';
export * from './imageGalleryContext/ImageGalleryContext';
export * from './keyboardContext/KeyboardContext';
export * from './messageContext/MessageContext';
export * from './messageInputContext/hooks/useCreateMessageInputContext';
export * from './messageInputContext/hooks/useMessageDetailsForState';
export * from './messageInputContext/MessageInputContext';
export * from './messageOverlayContext/MessageOverlayContext';
export * from './messagesContext/MessagesContext';
export * from './paginatedMessageListContext/PaginatedMessageListContext';
export * from './overlayContext/OverlayContext';
export * from './overlayContext/OverlayProvider';
export * from './suggestionsContext/SuggestionsContext';
export * from './themeContext/ThemeContext';
export * from './themeContext/utils/theme';
export * from './threadContext/ThreadContext';
export * from './toastContext/ToastContext';
export * from './translationContext/TranslationContext';
export * from './typingContext/TypingContext';
export * from './utils/getDisplayName';
