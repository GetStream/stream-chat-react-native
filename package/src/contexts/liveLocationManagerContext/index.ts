export * from './LiveLocationManagerContext';
export * from './hooks/useHandleLiveLocationEvents';
