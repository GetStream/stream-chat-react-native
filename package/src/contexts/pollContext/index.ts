export * from './createPollContentContext';
export * from './pollContext';
