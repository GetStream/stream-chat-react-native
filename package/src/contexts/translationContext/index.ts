export * from './TranslationContext';
export * from './types';
export * from './isDayOrMoment';
