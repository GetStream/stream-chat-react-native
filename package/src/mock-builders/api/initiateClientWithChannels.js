import { getOrCreateChannelApi } from './getOrCreateChannel';
import { useMockedApis } from './useMockedApis';

import { generateChannel } from '../generator/channel';
import { generateMember } from '../generator/member';
import { generateUser } from '../generator/user';
import { getTestClientWithUser } from '../mock';

const initChannelFromData = async ({ channelData, client, defaultGenerateChannelOptions }) => {
  const mockedChannelData = generateChannel({
    ...defaultGenerateChannelOptions,
    ...channelData,
  });

  useMockedApis(client, [getOrCreateChannelApi(mockedChannelData)]);
  const channel = client.channel(mockedChannelData.channel.type, mockedChannelData.channel.id);
  await channel.watch();
  jest.spyOn(channel, 'getConfig').mockImplementation(() => mockedChannelData.channel.config);
  // jest
  //   .spyOn(channel, 'getDraft')
  //   .mockImplementation(() => generateMessageDraft({ channel_cid: channel.cid }));
  return channel;
};

export const initiateClientWithChannels = async ({ channelsData, customUser } = {}) => {
  const user = customUser || generateUser();
  const client = await getTestClientWithUser(user);

  const defaultGenerateChannelOptions = {
    members: [generateMember({ user })],
  };
  const channels = await Promise.all(
    (channelsData ?? [defaultGenerateChannelOptions]).map((channelData) =>
      initChannelFromData({ channelData, client, defaultGenerateChannelOptions }),
    ),
  );

  return { channels, client };
};
