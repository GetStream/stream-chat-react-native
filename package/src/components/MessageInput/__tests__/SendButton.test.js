import React from 'react';

import { act, cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react-native';

import { OverlayProvider } from '../../../contexts';

import { initiateClientWithChannels } from '../../../mock-builders/api/initiateClientWithChannels';
import { Channel } from '../../Channel/Channel';
import { Chat } from '../../Chat/Chat';
import { SendButton } from '../SendButton';

const renderComponent = ({ client, channel, props }) => {
  return render(
    <OverlayProvider>
      <Chat client={client}>
        <Channel channel={channel}>
          <SendButton {...props} />
        </Channel>
      </Chat>
    </OverlayProvider>,
  );
};

describe('SendButton', () => {
  let client;
  let channel;

  beforeEach(async () => {
    const { client: chatClient, channels } = await initiateClientWithChannels();
    client = chatClient;
    channel = channels[0];
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();

    act(() => {
      channel.messageComposer.clear();
    });
  });

  it('should render a SendButton', async () => {
    const sendMessage = jest.fn();

    const props = { sendMessage };

    renderComponent({ channel, client, props });

    const { getByTestId, queryByTestId, toJSON } = screen;

    await waitFor(() => {
      expect(queryByTestId('send-button')).toBeTruthy();
      expect(sendMessage).toHaveBeenCalledTimes(0);
    });

    act(() => {
      fireEvent.press(getByTestId('send-button'));
    });

    await waitFor(() => {
      expect(sendMessage).toHaveBeenCalledTimes(1);
      expect(getByTestId('send-up')).toBeDefined();
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render a disabled SendButton', async () => {
    const sendMessage = jest.fn();

    const props = { disabled: true, sendMessage };

    renderComponent({ channel, client, props });

    const { getByTestId, queryByTestId, toJSON } = screen;

    await waitFor(() => {
      expect(queryByTestId('send-button')).toBeTruthy();
      expect(sendMessage).toHaveBeenCalledTimes(0);
    });

    act(() => {
      fireEvent.press(getByTestId('send-button'));
    });

    await waitFor(() => {
      expect(sendMessage).toHaveBeenCalledTimes(0);
      expect(getByTestId('send-right')).toBeDefined();
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  // TODO: Add it back once the command inject PR is merged
  it.skip('should show search button if the command is enabled', async () => {
    const sendMessage = jest.fn();

    const props = { sendMessage };

    channel.messageComposer.textComposer.setCommand({ description: 'Ban a user', name: 'ban' });

    renderComponent({ channel, client, props });

    const { queryByTestId } = screen;

    await waitFor(() => {
      expect(queryByTestId('search-icon')).toBeTruthy();
    });
  });
});
