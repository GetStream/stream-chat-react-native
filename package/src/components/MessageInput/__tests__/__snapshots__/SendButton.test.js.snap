// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendButton should render a non-editing disabled SendButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {},
    ]
  }
  testID="send-button"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#DBDBDB"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4292598747,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={16}
        cy={16}
        fill={
          {
            "payload": 4292598747,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r={16}
      />
      <RNSVGPath
        clipRule={0}
        d="M9.33398 14.6666H16.0007V9.33325L22.6673 15.9999L16.0007 22.6666V17.3333H9.33398V14.6666Z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`SendButton should render a non-editing enabled SendButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {},
    ]
  }
  testID="send-button"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#005FFF"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278214655,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={16}
        cy={16}
        fill={
          {
            "payload": 4278214655,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r={16}
      />
      <RNSVGPath
        clipRule={0}
        d="M14.6673 16V22.6667H17.334V16H22.6673L16.0007 9.33337L9.33398 16H14.6673Z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`SendButton should render an editing disabled SendButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {},
    ]
  }
  testID="send-button"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#DBDBDB"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4292598747,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={16}
        cy={16}
        fill={
          {
            "payload": 4292598747,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r={16}
      />
      <RNSVGPath
        clipRule={0}
        d="M9.33398 14.6666H16.0007V9.33325L22.6673 15.9999L16.0007 22.6666V17.3333H9.33398V14.6666Z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`SendButton should render an editing enabled SendButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {},
    ]
  }
  testID="send-button"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#005FFF"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278214655,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={16}
        cy={16}
        fill={
          {
            "payload": 4278214655,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r={16}
      />
      <RNSVGPath
        clipRule={0}
        d="M14.6673 16V22.6667H17.334V16H22.6673L16.0007 9.33337L9.33398 16H14.6673Z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
